OpenRA.Game/Network/UPnP.cs

```diff
@@ -10,127 +10,85 @@
 #endregion
 
 using System;
+using System.Collections.Generic;
+using System.Diagnostics;
 using System.Net;
-using Mono.Nat;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Open.Nat;
 
 namespace OpenRA.Network
 {
-	public static class UPnP
+	public class UPnP
 	{
-		static INatDevice natDevice;
+		static IEnumerable<NatDevice> natDevices;
+		static Mapping mapping;
 
-		public static void TryNatDiscovery()
-		{
-			try
-			{
-				NatUtility.Logger = Log.Channel("server").Writer;
-				NatUtility.Verbose = Game.Settings.Server.VerboseNatDiscovery;
-				NatUtility.DeviceFound += DeviceFound;
-				Game.Settings.Server.NatDeviceAvailable = false;
-				NatUtility.StartDiscovery();
-				Log.Write("server", "NAT discovery started.");
-			}
-			catch (Exception e)
-			{
-				Log.Write("server", "Can't discover UPnP-enabled device: {0}", e);
-				Game.Settings.Server.NatDeviceAvailable = false;
-				Game.Settings.Server.AllowPortForward = false;
-			}
-		}
+		public static IPAddress ExternalIP;
 
-		public static void StoppingNatDiscovery()
+		public static async Task TryNatDiscovery()
```

Comment:
Likewise here.  Please move the `Game.Settings` references to the part of `Game` that calls this instead.